backend/fastify-types.d.ts

```diff
@@ -0,0 +1,8 @@
+import { type FastifyRequest as OriginalFastifyRequest } from 'fastify';
+import { type JWTPayload } from 'jose';
+
+declare module 'fastify' {
+    interface FastifyRequest extends OriginalFastifyRequest {
+        user?: JWTPayload | null;
+    }
+}
```

Comment:
```suggestion
declare module 'fastify' {
    interface FastifyRequest  {
        user: HERE_SUPPOSED_TO_BE_USER_DTO;
    }
}
```
we don't have one yet, so probably leave just null for now

Reply:
I think we need to save `interface FastifyRequest extends OriginalFastifyRequest`. Otherwise, everything that we import from fastify has any type

Reply:
What do you think about it ?